ocamltest/main.ml

```diff
@@ -121,13 +120,9 @@ let get_test_build_directory_prefix test_dirname =
   if test_dirname = "." then root
   else Filename.concat root test_dirname
 
-let tests_to_skip = ref []
-
-let init_tests_to_skip () =
-  tests_to_skip := String.words (Sys.safe_getenv "OCAMLTEST_SKIP_TESTS")
-
 let test_file test_filename =
-  let skip_test = List.mem test_filename !tests_to_skip in
+  let tests_to_skip = String.words (Sys.safe_getenv "OCAMLTEST_SKIP_TESTS") in
+  let skip_test = List.mem test_filename tests_to_skip in
```

Comment:
Note that with this change, `tests_to_skip` is recomputed each time `test_file` is called. It would be better to compute it only once. A couple of ways which come to mind to do this:
1. Use a lazy at top-level : `let tests_to_skip = lazy (String.words (Sys.safe_getenv "OCAMLTEST_SKIP_TESTS"))`
2. Compute `tests_to_skip` outside of `test_file` and just pass it to `test_file` as an argument.